delivery_carrier_dangerous_goods_warning/models/dangerous_good_warning_mixin.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class DangerousGoodCarrierWarningMixin(models.AbstractModel):
+
+    _name = "dangerous.good.warning.mixin"
+    _description = "Display warning according to carrier and dangerous goods"
+
+    _line_field_name = None
+    _line_doc_m2o_field_name = None
+
+    contains_dangerous_goods = fields.Boolean(
+        compute="_compute_contains_dangerous_goods",
+        search="_search_contains_dangerous_goods",
+    )
+    display_dangerous_goods_carrier_warning = fields.Boolean(
+        compute="_compute_dangerous_goods_carrier_warning"
+    )
+    dangerous_goods_carrier_warning = fields.Text(
+        compute="_compute_dangerous_goods_carrier_warning"
+    )
+
+    def _compute_contains_dangerous_goods(self):
+        for doc in self:
+            doc.contains_dangerous_goods = any(
+                [l.product_id.is_dangerous_good for l in doc[self._line_field_name]]
```

Comment:
Why did you dropped the feature regarding adr_limited_amount_ids ? We need a way to be able to configure the type of dangerous goods a carrier is not allowed to transport.
Like DHL cannot ship "DG" goods but well "LG" goods

Reply:
Both sale exception and limited amounts are done as specific :wink: